### ISSUE TITLE
Libraries for Compiling

### ISSUE THREAD

**Issue by agowa** [ReikaKalseki/Reika_Mods_Issues#630]
Sorry for submitting an incident, but can someone give me a list of all dependencies required for building Dragon API, Rotarycraft and Reactorcraft?
There does not seem to be any information about it on the website, or at least I haven't found it...

Also there is a problem when I try to add the DragonAPI as a library to my environment it crashes with java.lang.NoSuchFieldError: field_76773_a
https://gist.github.com/agowa338/62b9ac183e4abc67fe38

**Comment by ReikaKalseki** [comment]
This is not a missing dependency issue, and the cause should be obvious to you if you have any background in MC modding.

Also, what are you trying to do?

**Comment by agowa** [comment]
At first I wanted to import your DragonAPI, Rotarycraft and Reactorcraft into IntelliJ to set some break points and figure out something about a damn crash. But I was tired of fixing all unresolved dependencies so I tried adding it simply as external library and found that there is a problem with deobfuscated sources, even tough codechickencore is present for translation...

**Comment by ReikaKalseki** [comment]
CCC only works for mods being loaded into the game, not other libraries.

**Comment by agowa** [comment]
Is there any other way of adding your mod to a dev environment (without recompiling as deobf)?

**Comment by ljfa-ag** [comment]
You could try this: https://github.com/Parker8283/BON2
It is a tool for deobfuscating mods, though I haven't tried it for libraries either

**Comment by agowa** [comment]
That would work for building, but it doesn't allow me to run in dev. Failing with this error: https://gist.github.com/agowa338/d2a2e59c7dc313f21edc

**Comment by ljfa-ag** [comment]
Afaik there is a way to make it ignore that, but I can't say it in public

**Comment by agowa** [comment]
uww, my mistake didn't delete the folder, but that still causes the selfcheck to fail...
So I think it's not the way to go.

Can't you simply provide a deobfuscated library so nobody would have to work around the self protection in the future? I don't say you should disable it, but make it so it still successes in deobfuscated environments.

**Comment by ReikaKalseki** [comment]
I have neither deobfuscated versions nor the means to make them.

**Comment by Aivech** [comment]
> I have neither deobfuscated versions nor the means to make them.

So downloading the relevant GitHub repos wouldn't work?

**Comment by ReikaKalseki** [comment]
That adds them as source, and you would need 30+ mods' APIs as dependencies.

**Comment by sclement41** [comment]
I have struggled with the same issue yesterday, and would appreciate any guidance. I am trying to use DragonAPI as a base library for my mod, as you suggest is possible on your site and on the Curse page. I admit I have limited experience with modding so far, but I am a longtime software developer. I have traced the missing field to be BiomeGenBase.biomeList, which is normally a private field on the BiomeGenBase class, hence the trouble finding it.

TL;DR; my question really is: is it possible to use DragonAPI as a base library for my mod, and if so, is it possible to use it while running my code in Eclipse/de-obfuscated environment, or do I have to build and obfuscate my mod for every time I want to run it, and have to forfeit in-editor debugging?

Thank you for your time and for the excellent work you put in these mods.

**Comment by ReikaKalseki** [comment]
For that you would need the access transformers. Contact me on the MC or FTB forums so I can give you a link to a archive containing all dependencies and the ATs.
